tests/spec/s-array.js

```diff
@@ -1568,4 +1568,301 @@ describe('Array', function () {
+            var actual = [0, 0, 0, 0, 0, -1];
+            actual.sort(descending);
+            expect(actual).toEqual([0, 0, 0, 0, 0, -1]);
+        });
+        it('sorting [f,e,d,a,c,b] descending should result in [f,e,d,c,b,a]', function () {
+            var actual = ['f', 'e', 'd', 'a', 'c', 'b'];
+            actual.sort(descending);
+            expect(actual).toEqual(['f', 'e', 'd', 'c', 'b', 'a']);
+        });
+        it('sorting [f,e,d,,,a,c,b] descending should result in [f,e,d,c,b,a,,,]', function () {
+            var actual = ['f', 'e', 'd', 1, 2, 'a', 'c', 'b'];
+            delete actual[3];
+            delete actual[4];
+            actual.sort(descending);
+            var expected = ['f', 'e', 'd', 'c', 'b', 'a'];
+            expected.length = 8;
+            expect(actual).toEqual(expected);
+        });
+        it('sorting [f,e,d,,null,,a,c,b] descending should result in [null,f,e,d,c,b,a,,,]', function () {
+            var actual = ['f', 'e', 'd', 1, null, 2, 'a', 'c', 'b'];
+            delete actual[3];
+            delete actual[5];
+            actual.sort(descending);
+            var expected = [null, 'f', 'e', 'd', 'c', 'b', 'a'];
+            expected.length = 9;
+            expect(actual).toEqual(expected);
+        });
+        it('sorting [f,e,d,undefined,null,,a,c,b] descending should result in [null,f,e,d,c,b,a,undefined,,]', function () {
+            var actual = ['f', 'e', 'd', undefined, null, 2, 'a', 'c', 'b'];
+            delete actual[5];
+            actual.sort(descending);
+            var expected = [null, 'f', 'e', 'd', 'c', 'b', 'a', undefined];
+            expected.length = 9;
+            expect(actual).toEqual(expected);
+        });
+        it('sorting [] descending should result in []', function () {
+            var actual = [];
+            actual.sort(descending);
+            expect(actual).toEqual([]);
+        });
+        it('sorting [1] descending should result in [1]', function () {
+            var actual = [1];
+            actual.sort(descending);
+            expect(actual).toEqual([1]);
+        });
+        it('sorted descending result should find only lesser or equal values while descending', function () {
+            testArr2.sort(descending);
+            for (var i = 0; i < testArr2.length - 1; i += 1) {
+                expect(testArr2[i] >= testArr2[i + 1]).toBe(true);
+            }
+        });
+        it('returned value should be source', function () {
+            var actual = [1, 3, 2];
+            expect(actual.sort()).toBe(actual);
+        });
+        it('is generic', function () {
+            var actual = {
+                0: 5,
+                1: 2,
+                2: 4,
+                3: 6,
+                4: 1,
+                5: 3
+            };
+            Array.prototype.sort.call(actual);
+            expect(Object.prototype.toString.call(actual)).toBe('[object Object]');
```

Comment:
what is this testing? if you want to ensure it's not an array, you could use `Array.isArray`, or, assert that `.not.toBe('[object Array]')`?